### PR TITLE
Setup oauth WIP

### PR DESCRIPTION
**Pull request by vlbee**
Adds signup, callback, logout  routes for oAuth via Github using Passport.js. 

Session management needs work 
Passport middleware needs to be modularised. 
HTTPS local host should be removed and tested without.

### CODE REVIEW

File: src/app.js [hunk 1]
```diff
@@ -1,8 +1,42 @@
...
 const exphbs = require('express-handlebars');
+const path = require('path');
+const cookieSession = require('cookie-session');
+const bodyParser = require('body-parser');
+
+require('dotenv').config();
+
+// Import Passport and Strategy
+const passport = require('passport');
+const Strategy = require('passport-github2').Strategy;
```
Comment (LawEKS) [inline src/app.js]: we could use object destructuring on 'Strategy'

File: .gitignore [hunk 2]
```diff
@@ -60,3 +63,5 @@ typings/
 # dotenv environment variables file
 *.env
 
+# other
+keys/
```
Comment (tdoran) [inline .gitignore]: Do we need keys/ twice in .gitignore?

**Comment by LawEKS** [comment]
When the server is run in this branch, this warning appears. 
`body-parser deprecated undefined extended: provide extended option src/app.js:51:20`

In the body-parser npm [docs](https://www.npmjs.com/package/body-parser#bodyparserurlencodedoptions) it explains why this warning appears, in the `bodyParser.urlencoded([options])` section

**Comment by helenzhou6** [comment]
Closed due to merge conflicts, that have been resolved and a new pull request made: #54

**Pull request closed without merging**